### DISCUSSION TITLE
Little Help needed in checkbox and selection design

### DISCUSSION THREAD

**Discussion by renrom** [Budibase/budibase#11492] [Help] [answered]
Hi, i have a custom query where i have to select multiple rows and do something with it afterwards . I am struggling to get this working.

What i need is something mentioned here:

https://github.com/Budibase/budibase/discussions/7453

And here:

https://github.com/Budibase/budibase/discussions/6548

In the first example, it looks like a table, but i doubt it is a table. i see the dataprovider, a repeater, a checkbox and text paragraphs. But how to manage that into a very nice design that it looks like a table (with a header)

how can i determine which rows are selected and put it into an array, knowing that i can use an id from the table. i think i need some javascript to build up and change the array after selecting or de-selecting rows, but i just need a little example. 

If someone can hand a simple example i would be gratefull!

**Comment by renrom** [comment]
Ok, i think i am getting there. but there seems to be a bug or something i miss. I create a litte app to debug and a mysql database with one test table. 

This is the layout, the dataprovider points to a custom query wich is a simple 'select * from selecttest':

![image](https://github.com/Budibase/budibase/assets/26928569/a5e5631d-36cf-4f02-b521-e9cf84c9271b)

As you see there are 3 rows. 

If i select the first or the second row, the last row also selects.. is that a bug?

![msedge_oQutASDQAA](https://github.com/Budibase/budibase/assets/26928569/beb723fa-56c5-4af2-9185-58d775a7f879)


This is selfhosted running 2.9.5

**Reply by ConorWebb96** [reply]
Hey @renrom,

You can't create dynamic form components within Budibase at the moment theres an open issue on this here; https://github.com/Budibase/budibase/issues/5658. The issue you are experiencing is because all the checkboxes are seen as the same field. There isn't a great way of differentiating between them, therefore I think the above issue likely needs fixed before you. would be able to easily do this.

I hope this makes sense.

**Reply by renrom** [reply]
So the other 2 threads i mentioned on my first post here should/are having the same issue then? So basicly, it is not possible to make a some sort of selectable table with a custom SQL query?

**Reply by ConorWebb96** [reply, marked as answer]
The table component has selectable functionality built in. So you could use that to get the selected items and then execute your custom query that way?

<img width="1465" alt="Screenshot 2023-08-11 at 12 16 33" src="https://github.com/Budibase/budibase/assets/126772285/0bafabdf-906c-4bcb-8b7e-a482cd93e325">

In the below example I'm getting the selected row _ids, fetching them from their specific table and then outputting the row in a notification. 

<img width="1159" alt="Screenshot 2023-08-11 at 12 26 12" src="https://github.com/Budibase/budibase/assets/126772285/5b7f0ae3-9f00-4612-8704-63ae04269ecc">

<img width="732" alt="Screenshot 2023-08-11 at 12 25 51" src="https://github.com/Budibase/budibase/assets/126772285/d645e856-f35b-42a6-b4bb-1a9b54b05388">

You likely could use a similar process but instead use the execute query action and bind up whatever necessary bindings you need to.

Hopefully this helps in some way.

**Reply by renrom** [reply]
Thanks for you reply Conor!, i realy appricate your extended answer, i is not realy an answer for my purpose, as far as i know the selectable option is not available with a custom query as a data source. The application i use for it, is more complicated :) hence the little example to show the checkbox thingy.  I do have a working situation though, but if this was working it would make some thing easier for me.

**Reply by renrom** [reply]
@ConorWebb96 , it looks like it works if i have the Form at another level, if i put it under the repeater it seems to work:

![image](https://github.com/Budibase/budibase/assets/26928569/a0312ef3-2fd3-47cc-93a3-822da102905f)

Result:

![msedge_tWzxHdFmBR](https://github.com/Budibase/budibase/assets/26928569/d4a83329-1360-4c01-94dc-6669075ffe9d)
